splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/MergeSortJoinStrategy.java

```diff
@@ -137,17 +139,16 @@ public static double mergeSortJoinStrategyLocalCost(CostEstimate innerCost, Cost
         double localLatency = config.getFallbackLocalLatency();
         double joiningRowCost = numOfJoinedRows * localLatency;
 
-        long innerTablePartitions = innerCost.partitionCount();
+        long outerTablePartitions = outerCost.partitionCount();
         double innerRowCount = innerCost.rowCount() > 1? innerCost.rowCount():1;
         int innerRowCountPerPartition =
-           (innerRowCount / innerTablePartitions) > Integer.MAX_VALUE ?
-           Integer.MAX_VALUE : (int)(innerRowCount / innerTablePartitions);
+           (innerRowCount / outerTablePartitions) > Integer.MAX_VALUE ?
```

Comment:
Can `(innerRowCount / outerTablePartitions) > Integer.MAX_VALUE` be ever `true`? What's the value of `innerRowCount / outerTablePartitions` then?

Reply:
innerRowCount is a double, which has a greater range than integer.  outerTablePartitions could be 1.  Not checking for max value could cause overflow.